usr/lib/common/pkey_utils.c

```diff
@@ -1681,16 +1749,26 @@ struct {                            \
     curve_type = get_cpacf_curve_type(pubkey->template);
     switch (curve_type) {
     case curve_ed25519:
+        if (sig_len != 2 * sizeof(edparam.ED25519.sig_r)) {
+            TRACE_ERROR("Signature has an invalid length of %ld bytes.\n", sig_len);
+            ret = CKR_ARGUMENTS_BAD;
+            goto done;
+        }
         fc = KDSA_EDDSA_VERIFY_ED25519;
         s390_flip_endian_32(edparam.ED25519.sig_r, sig);
         s390_flip_endian_32(edparam.ED25519.sig_s, sig + (sig_len / 2));
         s390_flip_endian_32(edparam.ED25519.pub, ecpoint);
         break;
     case curve_ed448:
+        if (sig_len != 2 * (sizeof(edparam.ED448.sig_r) - 7)) {
```

Comment:
Why `- 7` ?

Reply:
The 2 buffers for r and s each have 64 bytes, but an EdDSA signature has only 114 bytes. Similar to the 80 bytes buffers for p521, but signatures of 132 bytes.

Reply:
Ah ok, yes, makes sense. Please add a comment here to explain that, otherwise we will wonder in a few weeks again :-)

Reply:
ok, fixed all places and re-pushed.